analyses/immune-deconv/02-visualize_quantiseq.Rmd

```diff
@@ -260,30 +270,112 @@ ggplot(pdl1_cd8) +
   scale_color_identity() +
   labs(x = "Molecular subtype of sample",
-       y = "Log2 expression (FPKM) of PDL1") +
+       y = "Log2 expression (FPKM) of CD274") +
   ggpubr::theme_pubr() + 
   theme(axis.text.x = element_text(hjust = 1, 
                                    size = rel(0.6), 
-                                   angle = 90)) 
+                                   angle = 75)) 
 
 pdl1_distributions_plot
 
-ggsave(pdl1_distributions_plot_file, pdl1_distributions_plot, width = 8, height = 8)
+ggsave(pdl1_distributions_plot_file, pdl1_distributions_plot, width = 7, height = 5)
+
+```
+Visualize MB only, since that was the only significant result from the survival analyses
+
+```{r pdl1_mb, fig.width = 5, fig.height = 5}
+# select only mb samples
+pdl1_cd8_mb <- pdl1_cd8 %>%
+  filter(cancer_group_display == "Medulloblastoma")
+
+
+# Count group sizes to use in x-axis labels. Use this data frame going forward
+  df_counts <- pdl1_cd8_mb %>%
+    group_by(molecular_subtype) %>%
+    mutate(subtype_count = n()) %>%
+    ungroup() %>%
+    mutate(molecular_subtype = glue::glue("{molecular_subtype}\n(N = {subtype_count})"))
+  
+  wilcox_df <- ggpubr::compare_means(log2_expression ~ molecular_subtype, 
+                             data = df_counts,
+                             method = "wilcox.test")
+  
+# distributions of PDL1 expression
+cd274_plot_mb <- ggplot(pdl1_cd8_mb) + 
+  aes(x = molecular_subtype,
+      y = log2_expression) +
+  # remove outliers
+  geom_boxplot(outlier.shape = NA, color = "grey40", size = 0.2) + 
+  geom_jitter(width = 0.15, size = 0.75, alpha = 0.8, aes(color = "black")) + 
+  scale_color_identity() +
+  labs(x = "Molecular subtype of sample",
+       y = "Log2 expression (FPKM) of CD274") +
+  ggpubr::theme_pubr() + 
+  theme(axis.text.x = element_text(hjust = 1, 
+                                   size = rel(0.6), 
+                                   angle = 75)) 
+
+cd274_plot_mb
 
+ggsave(cd274_mb_plot_file, cd274_plot_mb, width = 5, height = 5)
 ```
 
+### Plot the ratio of CD8+/CD4+ T cells as a marker of "hot" tumor (potentially responsive to checkpoint blockade therapy)
+Not much here - only medulloblastoma samples seem to have more CD8+ cells than CD4+ cells
+
+```{r cd8_cd4 ratio, fig.width = 5, fig.height = 5}
+# subset for just cd8/cd4 ratio
+quantiseq_cd8 <- quantiseq_subset %>%
+  filter(cell_type == "cd8_cd4_ratio" & cancer_group != "Other")
+
+# drop inf
+quantiseq_cd8 <- do.call(data.frame,    # Replace Inf in data by NA
+                   lapply(quantiseq_cd8,
+                          function(x) replace(x, is.infinite(x), NA)))
+
+# drop NaN
+quantiseq_cd8 <- do.call(data.frame,    # Replace Inf in data by NA
+                   lapply(quantiseq_cd8,
+                          function(x) replace(x, is.nan(x), NA)))
+
+# remove NA rows from plot
+quantiseq_cd8 <- quantiseq_cd8 %>%
+  filter(!is.na(score))
+
+# distributions of PDL1 expression
+ratio_plot <- ggplot(quantiseq_cd8) + 
```

Comment:
I would recommend putting this on a log-scale; [this reference](https://clauswilke.com/dataviz/coordinate-systems-axes.html#nonlinear-axes) describes why ratios are more suitable on a log scale:

> Because multiplication on a log scale looks like addition on a linear scale, log scales are the natural choice for any data that have been obtained by multiplication or division. In particular, ratios should generally be shown on a log scale. 
> On a log scale, the value 1 is the natural midpoint, similar to the value 0 on a linear scale. We can think of values greater than 1 as representing multiplications and values less than 1 divisions. For example, we can write 10=1×10 and 0.1=1/10. The value 0, on the other hand, can never appear on a log scale. It lies infinitely far from 1. One way to see this is to consider that log(0)=−∞. Or, alternatively, consider that to go from 1 to 0, it takes either an infinite number of divisions by a finite value (e.g., 1/10/10/10/10/10/10⋯=0) or alternatively one division by infinity (i.e., 1/∞=0).


We can indicate in the text that 0's are necessarily not shown and potentially write up how many there were in a supp table. If we decide to add this to a table, code should be included here to calculate those numbers.

Reply:
This doesn't look super on a log scale, and it also seems to remove 109 more rows (not an error I get with the linear scale after NaN/Inf removal). I kept on linear because it seemed easier for people to think about what the ratios mean in terms of cell populations.. thoughts? It's not a stellar figure to begin with, and only signal in MB, hence the supplement.

Reply:
Current text for context:
>Finally, we asked whether any molecular subtypes might show an immunologically hot phenotype, as roughly defined by a greater proportion of CD8+ to CD4+ T cells and while we do see the highest ratios in Group 3 and Group 4 medulloblastomas (**Figure {@fig:Fig5}I**), very few tumors had ratios > 1, highlighting an urgent need to identify novel therapeutics for these immunologically cold pediatric brain tumors with poor prognosis.

Reply:
Given that so few are < 1, and the comparison between <1 and >1 ratios is the main need for log scale, keeping it on linear scale w/ this textual explanation is good with me!